### PR TITLE
Issue 894 rework failed instance status

### LINKED ISSUE #894: Rework FAILED instance status
Symptoms to fix:

1. Single-time network issue must not mark an instance as FAILED
2. The working job must not have a FAILED instance (misleading)

Fundumental questions:

1. Why do we need FAILED status?
3. What actions user must attempt in case of a FAILED instance?

### PR DESCRIPTION
**Pull request by TheBits**
[ ] refactor test_pool
[ ] add `dstack pool ps -w`
[ ] rename dstack pool show` -> `dstack pool ps`
[ ] add columns REGION, SPOT and CREATED
[ ] add locks

### CODE REVIEW

File: src/dstack/_internal/cli/commands/pool.py [hunk 1]
```diff
@@ -196,10 +207,22 @@ def _set_default(self, args: argparse.Namespace) -> None:
...
+            console.print(f" [bold]Pool name[/]  {resp.name}\n")
+            console.print(print_instance_table(resp.instances))
+            console.print()
+            return
+
+        try:
+            with Live(console=console, refresh_per_second=REFRESH_RATE_PER_SEC) as live:
+                while True:
+                    resp = self.api.client.pool.show(self.api.project, args.pool_name)
+                    live.update(print_instance_table(resp.instances))
```
Comment (Egor-S) [inline src/dstack/_internal/cli/commands/pool.py]: We don't print a pool name. Is it okay?

File: src/dstack/_internal/server/background/tasks/process_pools.py [hunk 2]
```diff
@@ -94,7 +94,15 @@ async def check_shim(instance_id: UUID) -> None:
...
                 )
                 instance.status = InstanceStatus.FAILED
-                await session.commit()
+
+            STARTING_TIMEOUT = 60 * 3
+            expire_starting = (
+                instance.created + timedelta(seconds=STARTING_TIMEOUT) < get_current_datetime()
+            )
+            if instance.status == InstanceStatus.STARTING and expire_starting:
+                instance.status = InstanceStatus.FAILED
```
Comment (Egor-S) [inline src/dstack/_internal/server/background/tasks/process_pools.py]: Do we keep using FAILED status?

**Comment by peterschmidt85** [comment]
`add dstack pool ps -w` is not critical at all

File: src/dstack/_internal/cli/commands/pool.py (hunk 1 above)
Comment (TheBits) [reply src/dstack/_internal/cli/commands/pool.py]: Fixed.

File: src/dstack/_internal/server/background/tasks/process_pools.py (hunk 2 above)
Comment (TheBits) [reply src/dstack/_internal/server/background/tasks/process_pools.py]: I've fixed all FAIL states.

**Pull request merged**